src/Common/Models/RobotModel.cs

```diff
@@ -155,4 +155,63 @@ public bool IsValid()
         }
         return true;
     }
+    /// <summary>
+    /// Set the topic to be enabled.
+    /// </summary>
+    /// <param name="topic"></param>
+    public void EnableRosTopic(RosTopicModel topic)
+    {
+        topic.Enabled = true;
+    }
+
+    /// <summary>
+    /// Set the topic to be disabled.
+    /// </summary>
+    /// <param name="topic"></param>
+    public void DisableRosTopic(RosTopicModel topic)
+    {
+        topic.Enabled = false;
+    }
+
+    /// <summary>
+    /// Get all topics of a robot in a single list.
+    /// </summary>
+    /// <param name="nodes"></param>
+    /// <returns></returns>
+    public List<RosTopicModel> GetAllRobotTopics()
+    {
+        List<RosTopicModel> topics = new List<RosTopicModel>();
```

Comment:
If you will use `HashSet` you will not have to worry about the items repeating in the collection. The `HashSet` contains only unique elements.

Reply:
Changed to hashset adapted return function type